logos-codegen/src/mir.rs

```diff
@@ -139,28 +154,33 @@ impl TryFrom<Hir> for Mir {
                             .into()
                         )
                     }
-                    RepetitionKind::ZeroOrOne => Ok(Mir::Maybe(Box::new(mir))),
-                    RepetitionKind::ZeroOrMore => Ok(Mir::Loop(Box::new(mir))),
-                    RepetitionKind::OneOrMore => {
+                    // 0 or 1
+                    (0, Some(1)) => Ok(Mir::Maybe(Box::new(mir))),
+                    // 0 or more
+                    (0, None) => Ok(Mir::Loop(Box::new(mir))),
+                    // 1 or more
+                    (1, None) => {
                         Ok(Mir::Concat(vec![mir.clone(), Mir::Loop(Box::new(mir))]))
                     }
-                    RepetitionKind::Range(range) => match range {
-                        RepetitionRange::Exactly(n) => {
+                    // Exact {n}
+                    (n, Some(m)) if m == n => {
                             let mut out = Vec::with_capacity(n as usize);
                             for _ in 0..n {
                                 out.push(mir.clone());
                             }
                             Ok(Mir::Concat(out))
```

Comment:
this and the following has a wrong indention. Not sure why rustfmt doesn't catch this.
Could you manually decremtn the ident of these three match arms?

Reply:
Nice catch!